src/DIRAC/ConfigurationSystem/Client/PathFinder.py

```diff
@@ -84,73 +106,148 @@ def getDatabaseSection(dbName, dbTuple=False, setup=False):
-  if not url:
-    return ""
-
-  # Trying if we are refering to the list of main servers
-  # which would be like dips://$MAINSERVERS$:1234/System/Component
-  # This can only happen if there is only one server defined
-  if ',' not in url:
-
-    urlParse = urlparse.urlparse(url)
-    server, port = urlParse.netloc.split(':')
-    mainUrlsList = []
-
-    if server == '$MAINSERVERS$':
-
-      # Operations cannot be imported at the beginning because of a bootstrap problem
-      from DIRAC.ConfigurationSystem.Client.Helpers.Operations import Operations
-      mainServers = Operations().getValue('MainServers', [])
-      if not mainServers:
-        raise Exception("No Main servers defined")
-
-      for srv in mainServers:
-        mainUrlsList.append(urlparse.ParseResult(scheme=urlParse.scheme, netloc=':'.join([srv, port]),
-                                                 path=urlParse.path, params='', query='', fragment='').geturl())
-      return ','.join(mainUrlsList)
-
-  if len(url.split("/")) < 5:
-    url = "%s/%s" % (url, serviceName)
-  return url
-
-
-def getServiceFailoverURL(serviceName, serviceTuple=False, setup=False):
-  if not serviceTuple:
-    serviceTuple = divideFullName(serviceName)
-  systemSection = getSystemSection(serviceName, serviceTuple, setup=setup)
-  url = gConfigurationData.extractOptionFromCFG("%s/FailoverURLs/%s" % (systemSection, serviceTuple[1]))
-  if not url:
-    return ""
-  if len(url.split("/")) < 5:
-    url = "%s/%s" % (url, serviceName)
-  return url
+  system, service = divideFullName(system, service)
+  resList = []
+  mainServers = None
+  systemSection = getSystemSection(system, setup=setup)
+
+  # Add failover URLs at the end of the list
+  failover = "Failover" if failover else ""
+  for fURLs in ["", "Failover"] if failover else [""]:
+    urlList = []
+    urls = List.fromChar(gConfigurationData.extractOptionFromCFG("%s/%sURLs/%s" % (systemSection, fURLs, service)))
+
+    # Be sure that urls not None
+    for url in urls or []:
+
+      # Trying if we are refering to the list of main servers
+      # which would be like dips://$MAINSERVERS$:1234/System/Component
+      if '$MAINSERVERS$' in url:
+        if not mainServers:
+          # Operations cannot be imported at the beginning because of a bootstrap problem
+          from DIRAC.ConfigurationSystem.Client.Helpers.Operations import Operations
+          mainServers = Operations(setup=setup).getValue('MainServers', [])
+        if not mainServers:
+          raise Exception("No Main servers defined")
+
+        for srv in mainServers:
+          _url = checkServiceURL(url.replace('$MAINSERVERS$', srv), system, service)
+          if _url not in urlList:
+            urlList.append(_url)
+        continue
+
+      _url = checkServiceURL(url, system, service)
+      if _url not in urlList:
+        urlList.append(_url)
+
+    # Randomize list if needed
+    resList.extend(List.randomize(urlList) if randomize else urlList)
```

Comment:
Is there ever a reason for not randomizing?

Reply:
Given that there are `failoverURLs` that are always added to the end of the list and thus you can always specify more priority servers, I see no reason for not randomizing. Ill change it